### ISSUE TITLE
cannot download and parse GEO files 

### ISSUE THREAD

**Issue by Mengflz** [guma44/GEOparse#62]
After I downloaded Series Matrix File(s), GEOparse.get_GEO function can't work and show there isn't series.
![image](https://user-images.githubusercontent.com/51097823/87120098-3eb15e00-c2b2-11ea-97eb-501d0bf6904f.png)

So I try to use GEOparse.get_GEO function to download files from website. It turned out that.
![image](https://user-images.githubusercontent.com/51097823/87120260-a071c800-c2b2-11ea-8380-c4bce4a92745.png)
It seems like url is wrong.

**Comment by guma44** [comment]
@Mengflz 
Hi, could you give complete snippet so I could look at it?

**Comment by Mengflz** [comment]
![image](https://user-images.githubusercontent.com/51097823/87301085-8863a800-c541-11ea-883d-fc4ba8fbc44b.png)
![image](https://user-images.githubusercontent.com/51097823/87301166-a7fad080-c541-11ea-8a67-41b3574fb103.png)
Here is the error message. When I download GSE52562,I got some problems. This dataset can be downloaded from GEO directly.

**Comment by guma44** [comment]
Could you also share version of python and GEOparse? For me it is working without any problems.

**Comment by Mengflz** [comment]
My GEOparse is version 2.0.1, and  my python version is  3.6.8

**Comment by daniwelter** [comment]
Is there any update on this issue? I have the same problem trying to get the metadata from 190 GEO series. Downloads fail during the checksum stage 
`Download failed due to 'Downloaded size do not match the expected size for ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE47nnn/GSE47598/soft/GSE47598_family.soft.gz'. ID could be incorrect or the data might not be public yet.`
I can download the files manually but there are a lot of them and the two I tested I couldn't open using GEOparse on local files either.
I use GEOparse v2.0.1 and python 3.8.

**Comment by carlosvega** [comment]
As a workaround you can do `export GEOPARSE_USE_HTTP_FOR_FTP=yes` before running your code.

**Comment by robertcv** [comment]
I have the same problem with GSE39582. I don't think it's GEOparser's fault because I am having the same problem with manually downloading the SOFT file. Using Firefox or `wget` I am constantly getting different sized files. Using HTTP instead of FTP (`export GEOPARSE_USE_HTTP_FOR_FTP=yes`) solves the issue.

**Comment by guma44** [comment]
Hi, sorry for not replying for long time. Some reason might be that you (might be) behind some corporate proxy. The option `GEOPARSE_USE_HTTP_FOR_FTP=yes` was introduced because FTP did not want to work with Travis CI. I will check the functions that @carlosvega mentioned but it is hard to debug these issues as for me everything is working.

**Comment by carlosvega** [comment]
Yes, but I was using the same VPN as @daniwelter but for me it was working, my guess is that is some network issue… if it doesn't work on the browser then is not your code fault. But maybe you could add the GEOPARSE_USE_HTTP_FOR_FTP as argparse option or as failover for FTP.

**Comment by daniwelter** [comment]
@carlosvega @guma44 I wasn't using a VPN at all but the FTP > HTTP switch worked for me either way.

**Comment by CholoTook** [comment]
> Is there any update on this issue? I have the same problem trying to get the metadata from 190 GEO series. Downloads fail during the checksum stage
> `Download failed due to 'Downloaded size do not match the expected size for ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE47nnn/GSE47598/soft/GSE47598_family.soft.gz'. ID could be incorrect or the data might not be public yet.`
> I can download the files manually but there are a lot of them and the two I tested I couldn't open using GEOparse on local files either.
> I use GEOparse v2.0.1 and python 3.8.

I saw this sometimes, but sometimes it went away... Not sure the problem, I suspect it's NCBI's side dropping connections
